From 7fb56cd4408858f336c955f3f93708eb933ed8d8 Mon Sep 17 00:00:00 2001
From: Steve Langasek <steve.langasek@canonical.com>
Date: Sat, 5 Nov 2011 11:15:26 -0700
Subject: [PATCH] Always make mk-build-deps packages arch-dependent when
 --arch is passed

When an --arch argument is passed to mk-build-deps, we should output an
architecture-dependent package even if there are no architecture-dependent
build-dependencies, so that cross-installing the package in a multiarch
environment does the right thing.
---
 scripts/mk-build-deps.pl |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/scripts/mk-build-deps.pl b/scripts/mk-build-deps.pl
index 61ebd3e..678ea47 100755
--- a/scripts/mk-build-deps.pl
+++ b/scripts/mk-build-deps.pl
@@ -321,9 +321,10 @@ sub build_equiv
     if ($opts->{depends} =~ /\[|\]/) {
 	$arch = 'any';
 
-	if (defined $opt_arch) {
-	    $args = "--arch=$opt_arch ";
-	}
+    }
+    if (defined $opt_arch) {
+	$args = "--arch=$opt_arch ";
+	$arch = $opt_arch;
     }
 
     open EQUIVS, "| equivs-build $args-"
-- 
1.7.5.4

